core/templates/services/exploration-improvements-task-registry.service.ts

```diff
@@ -0,0 +1,420 @@
+ * not hold onto them. Doing so would be extremely wasteful, since every state
+ * would be holding a reference to the exact same instance. Instead, we pass
+ * them explicitly as needed.
+ */
+class SupportingStateStats {
+  public readonly answerStats: readonly AnswerStats[];
+  public readonly cstPlaythroughIssues: readonly CstPlaythroughIssue[];
+  public readonly eqPlaythroughIssues: readonly EqPlaythroughIssue[];
+  public readonly misPlaythroughIssues: readonly MisPlaythroughIssue[];
+
+  constructor(
+      answerStats: readonly AnswerStats[] = [],
+      cstPlaythroughIssues: readonly CstPlaythroughIssue[] = [],
+      eqPlaythroughIssues: readonly EqPlaythroughIssue[] = [],
+      misPlaythroughIssues: readonly MisPlaythroughIssue[] = []) {
+    this.answerStats = [...answerStats];
+    this.cstPlaythroughIssues = [...cstPlaythroughIssues];
+    this.eqPlaythroughIssues = [...eqPlaythroughIssues];
+    this.misPlaythroughIssues = [...misPlaythroughIssues];
+  }
+}
+
+/**
+ * Container for the tasks of an exploration's state, coupled with references to
+ * their supporting statistics.
+ *
+ * The class provides a container-like interface for convenience and because it
+ * fits with the "container" model the class aims to fulfill.
+ */
+class StateTasks implements Iterable<ExplorationTask> {
+  public readonly hbrTask: HighBounceRateTask;
+  public readonly iflTask: IneffectiveFeedbackLoopTask;
+  public readonly ngrTask: NeedsGuidingResponsesTask;
+  public readonly siaTask: SuccessiveIncorrectAnswersTask;
+
+  public readonly supportingStateStats: SupportingStateStats;
+
+  constructor(
+      tasksByType: ReadonlyMap<ExplorationTaskType, ExplorationTask>,
+      supportingStateStats: SupportingStateStats = new SupportingStateStats()) {
+    this.hbrTask = <HbrTask> tasksByType.get(
+      ImprovementsConstants.TASK_TYPE_HIGH_BOUNCE_RATE);
+    this.iflTask = <IflTask> tasksByType.get(
+      ImprovementsConstants.TASK_TYPE_INEFFECTIVE_FEEDBACK_LOOP);
+    this.ngrTask = <NgrTask> tasksByType.get(
+      ImprovementsConstants.TASK_TYPE_NEEDS_GUIDING_RESPONSES);
+    this.siaTask = <SiaTask> tasksByType.get(
+      ImprovementsConstants.TASK_TYPE_SUCCESSIVE_INCORRECT_ANSWERS);
+
+    this.supportingStateStats = supportingStateStats;
+  }
+
+  public refresh(expStats: ExplorationStats): void {
+    this.hbrTask.refreshStatus(
+      expStats, this.supportingStateStats.eqPlaythroughIssues.length);
+    this.iflTask.refreshStatus(
+      this.supportingStateStats.cstPlaythroughIssues.length);
+    this.ngrTask.refreshStatus(
+      this.supportingStateStats.answerStats);
+    this.siaTask.refreshStatus(
+      this.supportingStateStats.misPlaythroughIssues.length);
+  }
+
+  *[Symbol.iterator](): Iterator<ExplorationTask> {
+    yield this.hbrTask;
+    yield this.iflTask;
+    yield this.ngrTask;
+    yield this.siaTask;
+  }
+
+  map<U>(fn: <T extends ExplorationTask>(t: T) => U): U[] {
```

Comment:
What is t? (It's not used, so just wondering if this is standard or if it should be something like `_` instead?)

Reply:
But `t` can be replaced with any arbitrary letter? If so, should we replace it with `_` to indicate "unused"?

Reply:
Done.